pkg/library/gatewayapi/types.go

```diff
@@ -19,7 +19,31 @@ func (a PolicyByCreationTimestamp) Less(i, j int) bool {
 	p1Time := ptr.To(a[i].GetCreationTimestamp())
 	p2Time := ptr.To(a[j].GetCreationTimestamp())
 	if !p1Time.Equal(p2Time) {
-		return ptr.To(a[i].GetCreationTimestamp()).Before(ptr.To(a[j].GetCreationTimestamp()))
+		return p1Time.Before(p2Time)
+	}
+
+	//  The policy appearing first in alphabetical order by "{namespace}/{name}".
+	return client.ObjectKeyFromObject(a[i]).String() < client.ObjectKeyFromObject(a[j]).String()
+}
+
+type PolicyByTargetRefKindAndCreationTimeStamp []Policy
+
+func (a PolicyByTargetRefKindAndCreationTimeStamp) Len() int      { return len(a) }
+func (a PolicyByTargetRefKindAndCreationTimeStamp) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+func (a PolicyByTargetRefKindAndCreationTimeStamp) Less(i, j int) bool {
+	targetRef1 := a[i].GetTargetRef()
+	targetRef2 := a[j].GetTargetRef()
+
+	// Compare kind first
+	if targetRef1.Kind != targetRef2.Kind {
+		return targetRef1.Kind < targetRef2.Kind
```

Comment:
Sorting alphabetically? I guess ATM we're lucky **G**ateway < **H**TTPRoute < **S**ervice.

Reply:
Updated to sort by Gateway, then HTTPRoute and then alphebetically